semantic_model_generator/generate_model.py

```diff
@@ -261,13 +284,17 @@ def comment_out_section(yaml_str: str, section_name: str) -> str:
 
         # Since this method parses a raw YAML string, we track whether we're in the section by the indentation level.
         # This is a pretty rough heuristic.
-        if in_section:
-            current_indent_level = len(line) - len(line.lstrip())
-            if current_indent_level <= section_indent_level and stripped_line:
-                in_section = False
-
-        if in_section:
-            comment_indent = " " * section_indent_level
+        current_indent_level = len(line) - len(line.lstrip())
```

Comment:
just including some quick changes to preserve indentation level when sections are commented out - makes it a bit easier for the user to preserve the correct indentation if uncommenting